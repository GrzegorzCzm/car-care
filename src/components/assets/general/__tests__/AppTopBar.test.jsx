import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import toJson from 'enzyme-to-json';

import ConnectedAppTopBar, { AppTopBar } from '../AppTopBar';


describe('AppTopBar', () => {
  const props = {
    toggleMenuAction: jest.fn(),
  };

  it('should renders correctly', () => {
    const wrapper = shallow(<AppTopBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call toggle menu function when click on menu', () => {
    const wrapper = shallow(<AppTopBar {...props} />);
    wrapper.find('.Menu').prop('onClick')();


    expect(props.toggleMenuAction).toHaveBeenCalled();
  });

  it('should dispatch action', () => {
    const mockStore = configureStore([]);
    const store = mockStore({
      navigation: {
        isMenuShown: false,
      },
    });

    const expectedActions = [{ type: 'TOGGLE_MENU' }];

    const wrapper = mount(
      <Provider store={store}>
        <ConnectedAppTopBar />
      </Provider>,
    );

    wrapper.find('AppTopBar').prop('toggleMenuAction')();

    expect(store.getActions()).toEqual(expectedActions);
  });
});
