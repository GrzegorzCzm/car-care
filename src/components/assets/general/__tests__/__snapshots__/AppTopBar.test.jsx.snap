// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTopBar should renders correctly 1`] = `
<WithStyles(ForwardRef(AppBar))
  color="default"
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <Memo([object Object])
      className="Menu"
      onClick={[MockFunction]}
    />
    <WithStyles(ForwardRef(Typography))
      color="inherit"
      variant="subtitle1"
    >
      Car Care
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
